test/quantization/core/test_quantized_tensor.py

```diff
@@ -267,23 +267,30 @@ def test_qtensor_legacy_new_failure(self):
         self.assertRaises(RuntimeError, lambda: qr.new(torch.Size([2, 3])))
         self.assertRaises(RuntimeError, lambda: qr.new([6]))
 
-    def test_per_channel_qtensor_creation(self):
+    def test_per_channel_qtensor_creation_cpu(self):
+        self._test_per_channel_qtensor_creation(torch.device('cpu'))
+
+    @unittest.skipIf(not TEST_CUDA, "No gpu is not available.")
+    def test_per_channel_qtensor_creation_cuda(self):
+        self._test_per_channel_qtensor_creation(torch.device('cuda'))
+
+    def _test_per_channel_qtensor_creation(self, device):
         numel = 10
         ch_axis = 0
-        scales = torch.rand(numel)
-        zero_points_int = torch.randint(0, 10, size=(numel,))
-        zero_points_float = torch.randn(numel)
+        scales = torch.rand(numel).to(device)
+        zero_points_int = torch.randint(0, 10, size=(numel,)).to(device)
+        zero_points_float = torch.randn(numel).to(device)
```

Comment:
Can we directly create these tensors on the device `torch.randn(.., device=device)`

Reply:
yeah, let me fix that. I think its fixed in the final diff, but it must not have been applied at this point in the stack.

Reply:
fixed